core/iwasm/libraries/debug-engine/debug_engine.c

```diff
@@ -971,6 +995,42 @@ wasm_debug_instance_remove_breakpoint(WASMDebugInstance *instance, uint64 addr,
     return true;
 }
 
+bool
+wasm_debug_instance_on_failure(WASMDebugInstance *instance)
+{
+    WASMExecEnv *exec_env;
+
+    if (!instance)
+        return false;
+
+    exec_env = bh_list_first_elem(&instance->cluster->exec_env_list);
```

Comment:
Should `os_mutex_lock(&instance->wait_lock)` firstly, in case that other threads are created or destroyed and instance->cluster->exec_env_list may be changed.

Refer to similar code in thread_manager.c:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/thread-mgr/thread_manager.c#L322

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/thread-mgr/thread_manager.c#L344